### PR TITLE
Fix retrieving last string in indefinite length containers

### PR DESCRIPTION
**Pull request by aggarg**
Description
------------
Indefinite length containers have a break character (0xFF) at the end to indicate the end of the container. When the last element of an indefinite length container was a string, the returned string was offset by one because of the break character.

This commit addresses the problem by adjusting the string to account for the break character in case the string happens to be the last element in an indefinite length container.

This solution was proposed by @qiutongs here: https://github.com/aws/amazon-freertos/pull/1955

Issue
-----
This issue was reported here: https://github.com/aws/amazon-freertos/issues/1915

Signed-off-by: Gaurav Aggarwal <aggarg@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c [hunk 1]
```diff
@@ -431,6 +431,14 @@ TEST( Serializer_Unit_CBOR, Encoder_map_nest_array )
     TEST_ASSERT_TRUE( cbor_value_at_end( &arrayElement ) );
 }
 
+/* JSON Representation:
```
Comment (aggarw13) [inline libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c:434]: (Nit): I have tried to use the "_CBOR diagnostic notation/representation_" words (as mentioned on cbor.me) instead of JSON to avoid any possible confusion.

File: libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c [hunk 2]
```diff
@@ -445,6 +453,201 @@ static const uint8_t _testEncodedNestedMap[] =
...
+ * [
+ *    "El1",
+ *    1,
+ *    "El3",
+ *    [
+ *       "Last"
+ *    ]
+ * ]
+ */
+static const uint8_t _testEncodedIndefiniteLengthArray[] =
```
Comment (aggarw13) [inline libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c:466]: `_testEncodedIndefiniteLengthArrayWithNestedArray`?

File: libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c [hunk 3]
```diff
@@ -445,6 +453,201 @@ static const uint8_t _testEncodedNestedMap[] =
...
+        0xFF,                                                                   /* # primitive(*) */
+    0xFF,                                                                       /* # primitive(*) */
+};
+
+/* JSON Representation:
+ * {
+ *     "result": h'AABBCCFF'
+ * }
+ */
+static const uint8_t _testEncodedIndefiniteLengthMapWithFFatTheEnd[] =
```
Comment (aggarw13) [inline libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c:627]: Should similar corner cases be tested for arrays as well to be comprehensive in our test coverage?

File: libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c (hunk 1 above)
Comment (aggarg) [reply libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c]: Done.

File: libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c (hunk 3 above)
Comment (aggarg) [reply libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c:627]: Done.

File: libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c (hunk 2 above)
Comment (aggarg) [reply libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c]: Done.
